crates/sui-package-management/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, bail};
+use std::path::PathBuf;
+
+use move_package::lock_file::{self, LockFile};
+use sui_json_rpc_types::{get_new_package_obj_from_response, SuiTransactionBlockResponse};
+use sui_sdk::wallet_context::WalletContext;
+
+pub enum LockCommand {
+    Publish,
+    Upgrade,
+}
+
+pub async fn update_lock_file(
+    context: &WalletContext,
+    command: LockCommand,
+    install_dir: Option<PathBuf>,
+    lock_file: Option<PathBuf>,
+    response: &SuiTransactionBlockResponse,
+) -> Result<(), anyhow::Error> {
+    let chain_identifier = context
+        .get_client()
+        .await?
+        .read_api()
+        .get_chain_identifier()
+        .await?;
+
+    let (original_id, version, _) = get_new_package_obj_from_response(response)
+        .ok_or_else(|| anyhow!("No package object response"))?;
+
+    let (install_dir, lock_file) = match (install_dir, lock_file) {
+        (Some(install_dir), Some(lock_file)) => (install_dir, lock_file),
+        (None, Some(lock_file)) => (PathBuf::from("."), lock_file),
+        (Some(_), None) => bail!("No lock file exists."),
+        // We need an install directory to have a working space for updating the lock file.
+        _ => bail!("Could not resolve install directory of move package."),
+    };
```

Comment:
Is there something that people can do if they see these errors? Something that we can get them to investigate, or a fix-up/clean-up step, or a place they can go to get help? At the moment it's just something they can cry about.